plugins/importtext_procedure.cc

```diff
@@ -801,9 +813,10 @@ struct ImportTextProcedureWorkInstance
                 }
 
                 if (config.autoGenerateHeaders) {
-                    stream.seekg(0);
-                    auto limit = fields.size();
-                    for (ssize_t i = 0; i < limit; ++i) {
+                    // Re-open stream
+                    stream.open(config.dataFileUrl, { { "mapped", "true" } });
```

Comment:
You can't seek all streams, but you can reopen them all.  The seek would fail on gzipped files; the reopen doesn't.  Generally you don't want to seek, or you want to only seek when you know it's possible and fall back on not seeking if not.